src/kloewe/dot/src/dot.h

```diff
@@ -0,0 +1,92 @@
+/*----------------------------------------------------------------------------
+  File    : dot.h
+  Contents: dot product (cpu dispatcher)
+  Author  : Kristian Loewe
+----------------------------------------------------------------------------*/
+#ifndef DOT_H
+#define DOT_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*----------------------------------------------------------------------------
+  Enum to encode the sets of dot implementations
+----------------------------------------------------------------------------*/
+enum flags {
+    DOT_NAIVE  = 1,
+    DOT_SSE2   = 2,
+    DOT_AVX    = 3,
+    #ifndef DOT_NOFMA
```

Comment:
You're right. Come to think about it, it doesn't even accomplish anything to hide DOT_AVXFMA there. The reason I introduced DOT_NOFMA was that Travis failed to build the avx-fma version. Apparently, gcc 4.6.3 doesn't support FMA. Strangely though, it doesn't complain about the flag (-mfma) and also defines __FMA__. But then it fails to find the relevant intrinsics (which are used if __FMA__ is defined) because it doesn't have the necessary header fmaintrin.h. So I introduced DOT_NOFMA and changed the CMakeLists.txt so that it defines DOT_NOFMA if gcc is too old to use the avx-fma version.